### PR TITLE
utils/hash_structs - hash_set_for_each

### PR DESCRIPTION
**Pull request by tsvisabo**

### CODE REVIEW

File: utils/hash_containers.c [hunk 1]
```diff
@@ -128,6 +128,19 @@ void hash_set_free(hash_set_t *set) {
   *set = NULL;
 }
 
+retcode_t hash_set_for_each(hash_set_t *set, hash_on_container_func func,
+                            void *container) {
+  retcode_t ret = RC_OK;
+  hash_set_entry_t *curr_entry = NULL;
```
Comment (th0br0) [inline utils/hash_containers.c:134]: As we call a `bitset` a `bitset`, we should probably call this `hashset` as well? (just a side remark)

File: utils/hash_containers.c [hunk 2]
```diff
@@ -128,6 +128,19 @@ void hash_set_free(hash_set_t *set) {
...
+  hash_set_entry_t *tmp_entry = NULL;
+  HASH_ITER(hh, *set, curr_entry, tmp_entry) {
+    if ((ret = func(container, curr_entry->hash))) {
```
Comment (th0br0) [inline utils/hash_containers.c]: This makes the implicit assumption that `RC_OK == 0`.

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: utils/hash_containers.c (hunk 2 above)
Comment (tsvisabo) [reply utils/hash_containers.c]: yeah, we can try to explicitly use `RC_OK`, should be enforced anywhere though

File: utils/hash_containers.c (hunk 1 above)
Comment (tsvisabo) [reply utils/hash_containers.c:134]: Sounds good for set, but then we have hashqueue,hashlist,hashdll (hashdoublelinkedlist?)

**Review by th0br0 — APPROVED** [review]

**Pull request merged**